### PR TITLE
App logs

### PR DESCRIPTION
**Pull request by ShamaUgale**
implemented logging using log4j

### CODE REVIEW

**Comment by ShamaUgale** [comment]
@anandbagmar please review

File: src/main/java/com/appium/filelocations/FileLocations.java [hunk 1]
```diff
@@ -21,5 +21,6 @@
     String APPIUM_LOGS_DIRECTORY = OUTPUT_DIRECTORY + "appiumlogs" + File.separator;
     String ADB_LOGS_DIRECTORY = OUTPUT_DIRECTORY + "adblogs" + File.separator;
     String DEVICE_LOGS_DIRECTORY = "deviceLogs" + File.separator;
+    String TEST_LOGS_DIRECTORY = "applicationTestLogs" + File.separator;
```
Comment (anandbagmar) [inline src/main/java/com/appium/filelocations/FileLocations.java]: Change applicationTestLogs to testLogs

File: src/main/java/com/cucumber/listener/CucumberScenarioListener.java [hunk 2]
```diff
@@ -119,14 +118,24 @@ private void runStartedHandler(TestRunStarted event) {
...
         LOGGER.info("caseStartedHandler: " + scenarioName);
         Integer scenarioRunCount = getScenarioRunCount(scenarioName);
+        String normalisedScenarioName = normaliseScenarioName(scenarioName);
+        String testLogFileName= FileLocations.REPORTS_DIRECTORY
+                + normalisedScenarioName
+                + File.separator
+                + FileLocations.TEST_LOGS_DIRECTORY
+                + "/run-"+ scenarioRunCount;
+        testExecutionContext.addTestState("testLog", testLogFileName);
+        System.setProperty("log_dir", testLogFileName);
```
Comment (anandbagmar) [inline src/main/java/com/cucumber/listener/CucumberScenarioListener.java]: Remove? If required, change to LOG_DIR

File: src/test/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# root logger
+log4j.rootLogger=INFO,CONSOLE,R,HTML,TTCC
```
Comment (anandbagmar) [inline src/test/resources/log4j.properties]: Remove HTML, TTCC

File: src/test/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# appender
+log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
+log4j.appender.R=org.apache.log4j.RollingFileAppender
+
+#log file location
+log4j.appender.R.File=./${log_dir}/testlog.log
```
Comment (anandbagmar) [inline src/test/resources/log4j.properties]: Change to LOG_DIR

**Review by anandbagmar — COMMENTED** [review]
Added review comments.

Also we need to update README with instructions about LOG_DIR

File: src/test/resources/log4j.properties (hunk 3 above)
Comment (ShamaUgale) [reply src/test/resources/log4j.properties]: done

File: src/test/resources/log4j.properties (hunk 4 above)
Comment (ShamaUgale) [reply src/test/resources/log4j.properties]: done

File: src/main/java/com/cucumber/listener/CucumberScenarioListener.java (hunk 2 above)
Comment (ShamaUgale) [reply src/main/java/com/cucumber/listener/CucumberScenarioListener.java]: removed. This is not needed.

File: src/main/java/com/appium/filelocations/FileLocations.java (hunk 1 above)
Comment (ShamaUgale) [reply src/main/java/com/appium/filelocations/FileLocations.java]: done

**Pull request merged**